src/sagemaker/estimator.py

```diff
@@ -1318,6 +1318,11 @@ def register(
         Returns:
             str: A string of SageMaker Model Package ARN.
         """
+        if isinstance(self.sagemaker_session, PipelineSession):
+            raise TypeError(
+                "estimator.register does not support PipelineSession at this moment. "
+                "Please use model.register under PipelineSession if you're using the ModelStep."
```

Comment:
nit: Please use `model.register` with `PipelineSession` ?

Reply:
Updated in the new commit